src/main/java/io/openapiprocessor/maven/ProcessMojo.java

```diff
@@ -53,7 +51,8 @@ public void execute () throws MojoExecutionException {
             String relativeSource = stripBaseDir (source.getAbsolutePath ());
             getLog().info(String.format ("%10s - ${project.basedir}/%s", "apiPath", relativeSource));
 
-            String targetDir = (String) properties.get (TARGET_DIR);
+            String targetDir = (String) properties.computeIfAbsent (TARGET_DIR, k -> project.getBuild().getDirectory() + "/generated-sources/" + id);
+            project.addCompileSourceRoot(targetDir);
```

Comment:
I like that  :-)

Reply:
Maybe the execution id should also be added to the targetDir?

Reply:
Right, modifying the explicitly set `apiPath` would be confusing.

Reply:
One last thing. I think it should use `File.separator` instead of the hard coded slash in line 54.

I already had a few annoying path separator issues in the processor which were fixed by consistently using File.separator.

Reply:
One more ;-) 

while looking at the plugin docs I noticed that this would not handle `openapi-processor-json`. It generates a json   resource file from the OpenAPI. (limited and bit outdated though).

Can we check the lifecycle phase and then add the sources (if `<phase>generate-sources</phase>`) or the resource (if `<phase>generate-resources</phase>`)?

Reply:
The mojo can easily be  executed at another stage by specifying that in the POM. 
I see three solutions:
1. The parameter could either be configurable, with `true` as default. Then document that for the json processor, the parameter should be set to `false`
2. add to compiler source, `if id != "json"` 
3. Ask the processor whether to add to the compiler source. This would require extending the Service interface of `io.openapiprocessor.api.v2.OpenApiProcessor`

Reply:
I see.

I have changed it a bit to use an explicit option to disable adding the compile source root.